echopype/echodata/sensor_ep_version_mapping/v05x_to_v06x.py

```diff
@@ -0,0 +1,1131 @@
+    information) for the EK80 sensor only.
+
+
+    Parameters
+    ----------
+    ed_obj : EchoData
+        EchoData object that was created using echopype version 0.5.x.
+    sensor : str
+        The sensor used to create the v0.5.x file.
+
+    Notes
+    -----
+    The function directly modifies the input EchoData object.
+    """
+
+    set_groups_cls = SONAR_MODELS[sensor]["set_groups"]
+
+    _beam_groups_to_convention(ed_obj, set_groups_cls)
+
+    # add beam_group coordinate and beam_group_descr variable
+    num_beams = len(ed_obj._tree["Sonar"].children)
+    set_groups_cls._beamgroups = set_groups_cls.beamgroups_possible[:num_beams]
+    beam_groups_vars, beam_groups_coord = set_groups_cls._beam_groups_vars(set_groups_cls)
+
+    ed_obj["Sonar"] = ed_obj["Sonar"].assign_coords(beam_groups_coord)
+    ed_obj["Sonar"] = ed_obj["Sonar"].assign(**beam_groups_vars)
+
+    # add sonar_serial_number to EK80 Sonar group
+    if sensor == "EK80":
+
+        ed_obj["Sonar"] = ed_obj["Sonar"].assign(
+            {
+                "sonar_serial_number": (
+                    ["channel"],
+                    np.full_like(ed_obj["Sonar"].frequency_nominal.values, np.nan),
+                )
+            }
+        )
+
+    # TODO: Do we need to make attributes into variables
+    #  for EK60 and AZFP #681?
+
+
+def _move_transducer_offset_vars(ed_obj, sensor):
+    """
+    Moves transducer_offset_x/y/z from beam groups to Platform
+    group for EK60 and EK80. If more than one beam group exists,
+    then the variables are first collected and then moved to
+    Platform. Additionally, adds ``frequency_nominal`` to
+    Platform for the EK80 sensor.
+
+    Parameters
+    ----------
+    ed_obj : EchoData
+        EchoData object that was created using echopype version 0.5.x.
+    sensor : str
+        Variable specifying the sensor that created the file.
+
+    Notes
+    -----
+    The function directly modifies the input EchoData object.
+    """
+
+    if sensor in ["EK60", "EK80"]:
+        full_transducer_vars = {"x": [], "y": [], "z": []}
+
+        # collect transducser_offset_x/y/z from the beam groups
+        for beam_group in ed_obj._tree["Sonar"].children:
+            for spatial in full_transducer_vars.keys():
+                full_transducer_vars[spatial].append(beam_group.ds["transducer_offset_" + spatial])
+
+                # remove transducer_offset_x/y/z from the beam group
+                beam_group.ds = beam_group.ds.drop("transducer_offset_" + spatial)
+
+        # transfer transducser_offset_x/y/z to Platform
+        for spatial in full_transducer_vars.keys():
+            ed_obj["Platform"]["transducer_offset_" + spatial] = xr.concat(
+                full_transducer_vars[spatial], dim="channel"
+            )
+
+    if sensor == "EK80":
+        ed_obj["Platform"]["frequency_nominal"] = ed_obj["Vendor"].frequency_nominal.sel(
+            channel=ed_obj["Platform"].channel
+        )
```

Comment:
My previous comment might have been washed out, but I am thinking that now we have everything aligned with `channel`, do we still need `frequency_nominal` to be in this dataset? I can see this makes EK80 consistent with EK60 though, maybe that's the rationale here?
I am fine with keeping it here since it has a small footprint.

Reply:
In previous issues, we explicitly state that we should add `frequency_nominal` to every dataset that has `channel`. I think this is a good idea because then you don't have to go looking for what frequency corresponds to the channel. Furthermore, I think `Platform.frequency_nominal` is referenced in downstream processes. I highly suggest that we do not remove this.

Reply:
I agree with @b-reyes . If `channel` is present, so should `frequency_nominal`

Reply:
Sounds good.